src/metrics.rs

```diff
@@ -2578,7 +2578,7 @@ impl GooseAttack {
             match message.unwrap() {
                 GooseMetric::Request(request_metric) => {
                     // If there was an error, store it.
-                    if !request_metric.error.is_empty() {
+                    if !request_metric.success {
```

Comment:
This isn't enough -- if we remove this check here, we'll need to add it before we log the error text (as `success` can be `false` but `error` also empty and with this change you've made we could get "empty" errors, a regression to this functionality):
https://github.com/tag1consulting/goose/blob/7121f4617c8a07ebf7183a7cf67b6105bf0f7402/src/metrics.rs#L2562

Reply:
Will this work?

https://github.com/tag1consulting/goose/pull/394/files#diff-9f90e93b5696724fd8e4407c9db0bfd381d5468bcfbf2f09c17ac8510a85e104L2479-L2480

Reply:
IMO we shouldn't be making this change/fix in this PR as it's unrelated. Let's tackle this as a separate PR -- and in that case, I'd do what you've done here first, and then within the invoked function I'd add the second check only before where we care about if `error` isn't empty.

Reply:
Agreed. Removed.